libs/ledger/include/ledger/dag/dag_node.hpp

```diff
@@ -41,10 +42,13 @@ struct DAGNode
   using ConstByteArray = byte_array::ConstByteArray;
   using Digest         = ConstByteArray;
   using Signature      = ConstByteArray;
-  using DigestList     = std::vector<Digest>;
+  using DAGHashList    = std::vector<DAGHash>;
   using HasherType     = crypto::SHA256;
 
-  DAGNode()                   = default;
+  DAGNode()
+  {
+    hash.type = DAGHash::Type ::NODE;
```

Comment:
Worth make DAg Hash constructible via just a node type then this can be moved to init. list?

Reply:
I've added a constructor for building from node type. Also updated the no argument constructor.